perf/lua/column_scan_module.c

```diff
@@ -97,7 +98,51 @@ sum_iterator_rv_lua_func(struct lua_State *L)
 }
 #endif /* defined(ENABLE_READ_VIEW) */
 
-#if defined(ENABLE_SCANNER)
+#if defined(ENABLE_ARROW_AND_SCANNER)
+static int
+sum_arrow_lua_func(struct lua_State *L)
+{
+	uint32_t space_id = luaL_checkinteger(L, 1);
+	uint32_t index_id = luaL_checkinteger(L, 2);
+	uint32_t field_no = luaL_checkinteger(L, 3);
+	char key[8];
+	char *key_end = mp_encode_array(key, 0);
+	uint32_t fields[] = {field_no};
+	uint32_t field_count = lengthof(fields);
+	box_arrow_options_t *options = box_arrow_options_new();
+	box_arrow_options_set_batch_row_count(options, 4096);
+	struct ArrowArrayStream *stream = xmalloc(sizeof(*stream));
```

Comment:
Nit: No need to allocate it from the heap. You could store it on the stack. Same comment to the read view version.

Reply:
Made it on-stack